src/NUnitFramework/framework/Constraints/ExactCountConstraint.cs

```diff
@@ -31,9 +31,19 @@ namespace NUnit.Framework.Constraints
     /// item in a collection, succeeding only if a specified
     /// number of items succeed.
     /// </summary>
-    public class ExactCountConstraint : PrefixConstraint
+    public class ExactCountConstraint : Constraint
```

Comment:
The dual logic also bothers me. We have the if/else in `ApplyTo` and the final expression in the `Description`. I like @CharliePoole's idea of producing one of the two constraints based on context, but like him, I am not in the code at the moment :smile: